Sources/ArcGISToolkit/Components/Offline/OfflineMapViewModel.swift

```diff
@@ -48,123 +80,111 @@ class OfflineMapViewModel: ObservableObject {
         portalItemID = onlineMap.item!.id!
     }
     
-    /// Gets the preplanned map areas from the offline map task and loads the map models.
-    func loadPreplannedMapModels() async {
-        if offlineMapTask.loadStatus != .loaded {
-            try? await offlineMapTask.retryLoad()
-        }
-        
-        // Reset flag.
-        isShowingOnlyOfflineModels = false
+    /// The function called when a downloaded preplanned map area is removed.
+    func onRemoveDownloadOfPreplannedArea() {
+        // Delete the saved map info if there are no more downloads for the
+        // represented online map.
+        guard !hasDownloadedMapAreas else { return }
         
-        preplannedMapModels = await Result { @MainActor in
-            do {
-                return try await offlineMapTask.preplannedMapAreas
-                    .filter { $0.portalItem.id != nil }
-                    .sorted(using: KeyPathComparator(\.portalItem.title))
-                    .map {
-                        PreplannedMapModel(
-                            offlineMapTask: offlineMapTask,
-                            mapArea: $0,
-                            portalItemID: portalItemID,
-                            preplannedMapAreaID: $0.portalItem.id!,
-                            onRemoveDownload: onRemoveDownloadOfPreplannedArea(withID:)
-                        )
-                    }
-            } catch {
-                // If not connected to the internet, then return only the offline models.
-                if let urlError = error as? URLError,
-                   urlError.code == .notConnectedToInternet {
-                    isShowingOnlyOfflineModels = true
-                    return await loadOfflinePreplannedMapModels()
-                } else {
-                    throw error
-                }
-            }
+        OfflineManager.shared.removeMapInfo(for: portalItemID)
+    }
+    
+    var hasAnyPreplannedMapAreas: Bool {
```

Comment:
```suggestion
    private var hasPreplannedMapAreas: Bool {
```
Could be placed above the init and have doc added.